### PR TITLE
Cross compile link args order

### PR DESCRIPTION
**Pull request by mgautierfr**

### CODE REVIEW

**Comment by nirbheek** [comment]
Please add a description of what the specific problem was and how this solves it.

File: mesonbuild/backend/ninjabackend.py [hunk 1]
```diff
@@ -1274,12 +1274,12 @@ def generate_dynamic_link_rules(self, outfile):
                         pass
                 rule = 'rule %s%s_LINKER\n' % (langname, crstr)
                 if mesonlib.is_windows():
-                    command_template = ''' command = {executable} @$out.rsp
+                    command_template = ''' command = {linker_exelist} @$out.rsp
```
Comment (msink) [inline mesonbuild/backend/ninjabackend.py]: Why? This breaks build on windows.

Comment (mgautierfr) [inline mesonbuild/backend/ninjabackend.py]: Oups, my bad, I started to name it {linker_exelist} and renamed it {executable} latter.

I've missed this line in my 'git rebase -i'. I will fix it.

**Comment by mgautierfr** [comment]
> Please add a description of what the specific problem was and how this solves it.

On my cross_file for cross-compilation to windows using mingw, I have those properties:
`c_link_args = ['-lwinmm', '-lws2_32', '-lshlwapi', '-lrpcrt4']`
I need to link to those libraries as they provide some standard symbols for windows.

With the present link target generation we've got :
`[linker] -lwinmm -lws2_32 -lshlwapi -lrpcrt4 $ARGS [output_args] $in $LINK_ARGS $aliasing`
gcc search for symbol in libraries declare after where they are used and as the `-l...` are declared before the `$in $LINK_ARGS`, gcc doesn't find the symbol even if they are present.
With the new order, we've got :
`[linker] $ARGS [output_args] $in $LINK_ARGS -lwinmm -lws2_32 -lshlwapi -lrpcrt4 $aliasing`
and gcc can found the symbols.

> Please add a description …

Where ? Is it enough here or do you prefer I add it to the commit message ?

**Comment by nirbheek** [comment]
Here is ok, just add a link to this PR to the commit message, thanks!

**Comment by mgautierfr** [comment]
Hi,

Any clues if (and when) this PR be merged ?
Is it possible to have a release with this fix ? My cross compilation script is blocked without this fix and I always need to install meson from my custom repo branch. With a new release a simple `pip install meson` would be enough.

Thanks.

**Comment by nirbheek** [comment]
It will be merged as soon as @jpakkane gets to it. It will definitely be in 0.39.0 which will be next month, but we might also do a 0.38.2 point release because there's a bunch of other bugs that we've discovered since the 0.38.1 release.

**Pull request merged**